### PR TITLE
FIX #11687 issue add unit test

### LINKED ISSUE #11687: Add unit test for shardingsphere-shadow-core
Hi community,

This issue is for #11661

## Aim
Add unit test for shardingsphere-shadow-core project to test its public functions.

## Basic Qualifications
- Java  
- Maven  
- Junit.Test

## Classes
Add unit test for classes to test its public functions.

- [ ] ShadowDataSourceConfigurationYamlSwapper
- [ ] ShadowTableConfigurationYamlSwapper
- [ ] ShadowRuleAlgorithmProviderConfigurationYamlSwapper

## Examples FYI
- ShadowRuleConfigurationYamlSwapperTest
- YamlRootRuleConfigurationsForYamlShadowRuleConfigurationTest

### PR DESCRIPTION
**Pull request by hoorf**
Fixes #11687.

add unit test for :

ShadowDataSourceConfigurationYamlSwapper
ShadowTableConfigurationYamlSwapper
ShadowRuleAlgorithmProviderConfigurationYamlSwapper

### CODE REVIEW

**Comment by hoorf** [comment]
repeat pull

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/swapper/ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+public final class ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest {
+
+    private AlgorithmProvidedShadowRuleConfiguration algorithmProvidedShadowRuleConfiguration;
+
+    private YamlShadowRuleConfiguration yamlShadowRuleConfiguration;
+
+    private void buildAlgorithmProvidedShadowRuleConfiguration() {
```
Comment (menghaoranss) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/swapper/ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest.java:44]: Please put `private` methods under the `public` methods that reference them.

**Review by menghaoranss — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/swapper/ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest.java (hunk 1 above)
Comment (hoorf) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/swapper/ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest.java:44]: check style did not found this rule, is that check style rule file has update?

Comment (menghaoranss) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/swapper/ShadowRuleAlgorithmProviderConfigurationYamlSwapperTest.java]: No rule in check style, just more friendly for reading.

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**